src/paradigma/tremor/feature_extraction.py

```diff
@@ -347,83 +217,112 @@ def extract_tremor_power(
-            sensor_col = df_windowed[col], 
-            sampling_frequency = config.sampling_frequency,
-            window_type = config.window_type, 
-            segment_length_s = config.segment_length_s_psd,
-            overlap_fraction = config.overlap_fraction,
-            spectral_resolution = config.spectral_resolution_psd
-            )
-        df_windowed[f'{col}_spectrogram'] = signal_to_spectrogram(
-            sensor_col = df_windowed[col], 
-            sampling_frequency = config.sampling_frequency,
-            window_type = config.window_type, 
-            segment_length_s = config.segment_length_s_mfcc,
-            overlap_fraction = config.overlap_fraction
-            )
-    
-    # compute the total PSD and spectrogram (summed across the 3 gyroscope axes)
-    df_windowed['total_PSD'] = df_windowed[[f"{y}_PSD" for y in config.gyroscope_cols]].sum(axis=1)
-    df_windowed['total_spectrogram'] = df_windowed[[f"{y}_spectrogram"for y in config.gyroscope_cols]].sum(axis=1) 
-
-    # compute the cepstral coefficients
-    mfcc_cols = generate_mel_frequency_cepstral_coefficients(
-        spectrogram=df_windowed['total_spectrogram'],
-        segment_length_s = config.segment_length_s_mfcc,
-        sampling_frequency=config.sampling_frequency,
-        fmin=config.fmin_mfcc,
-        fmax=config.fmax_mfcc,
-        n_filters=config.n_dct_filters_mfcc,
-        n_coefficients=config.n_coefficients_mfcc
-        )
-    df_windowed = pd.concat([df_windowed, mfcc_cols], axis=1)
-
-    # compute the frequency of the peak in the PSD
-    df_windowed['freq_peak'] = extract_frequency_peak(
-        total_psd = df_windowed['total_PSD'],
-        freq_vect = df_windowed['gyroscope_x_freqs_PSD'],
-        fmin = config.fmin_peak,
-        fmax = config.fmax_peak
-    )
-    
-    # compute the low frequency power (for detection of slow arm movement)
-    df_windowed['low_freq_power'] = extract_low_freq_power(
-        total_psd = df_windowed['total_PSD'],
-        freq_vect = df_windowed['gyroscope_x_freqs_PSD'],
-        fmin = config.fmin_low_power,
-        fmax = config.fmax_low_power,
-        spectral_resolution = config.spectral_resolution_psd
+
+def extract_spectral_domain_features(config, data):
+    """
+    Compute spectral domain features from the gyroscope data.
+
+    This function computes Mel-frequency cepstral coefficients (MFCCs) based on the spectrogram, 
+    and computes the frequency of the peak, the tremor power, and the non-tremor power based on 
+    the power spectral density of the windowed gyroscope data.
+
+    Parameters
+    ----------
+    config : object
+        Configuration object containing settings such as sampling frequency, window type, 
+        and MFCC parameters.
+    windowed_data : numpy.ndarray
+        A 2D numpy array where each row corresponds to a window of gyroscope data.
+
+    Returns
+    -------
+    dict
```

Comment:
It actually seems to return a DataFrame (if you specify this in the function: `function(a: np.ndarray) -> pd.DataFrame():` it will raise an error during the testing.